torch/csrc/jit/serialization/unpickler.cpp

```diff
@@ -868,6 +868,23 @@ void Unpickler::rebuildTensor(bool quantized) {
     impl->set_storage_offset(storage_offset);
     impl->set_sizes_and_strides(size, stride);
     result = autograd::make_variable(result, requires_grad);
+
+    // Handle if math_bits were pickled.
+    // See `args` of _reduce_ex_internal
+    // for a regular tensor (final else case).
+    // NOTE: `math_bits` is the 7th arg.
+    // NOTE: This is only meant for regular tensor and not quantized!
+    if (!quantized) {
+      bool has_math_bits = elements.size() == 7;
+      // Tensors pickled before this patch didn't
+      // have this argument for storing MathBits,
+      // in that case, we do nothing.
+      if (has_math_bits) {
+        auto math_bits = elements.at(idx + 2).toGenericDict();
```

Comment:
why idx + 2?

Reply:
`idx + 1` points to backward hooks (will add a comment).

https://github.com/pytorch/pytorch/blob/0ae4ac2b6cfd7c35637c3dcd49e26536d0d9b08a/torch/csrc/jit/serialization/unpickler.cpp#L865